packages/react-native/ReactCommon/react/nativemodule/microtasks/IdleDeadline.cpp

```diff
@@ -0,0 +1,57 @@
+/*
+* Copyright (c) Meta Platforms, Inc. and affiliates.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+#include <react/renderer/runtimescheduler/RuntimeSchedulerBinding.h>
+#include <react/renderer/runtimescheduler/RuntimeScheduler.h>
+#include "IdleDeadline.h"
+#include <chrono>
+
+namespace facebook::react {
+
+IdleDeadline::IdleDeadline(
+    bool didTimeout,
+    RuntimeSchedulerTimePoint deadline)
+    : didTimeout_(didTimeout), deadline_(deadline) {}
+
+jsi::Value IdleDeadline::get(
```

Comment:
I'm not a fan of host objects because they generally require optimizations by hand to prevent inefficiencies. For example, in this case, we have to use dynamic property dispatch, and every time we call `timeRemaining()` we're creating and calling a new host function. Instead of trying to optimize that using caching, etc. it might just be easier to return a regular JS object with a pre-defined property `didTimeout` with the value we already have, and another property containing the host function you're creating here.